ui/keyboardMove/src/plugins/keyboardMove.ts

```diff
@@ -22,6 +23,74 @@ interface SubmitOpts {
 }
 type Submit = (v: string, submitOpts: SubmitOpts) => void;
 
+const speechLookup = new Map<string, string>([
+  ['a', 'a'],
+  ['b', 'b'],
+  ['c', 'c'],
+  ['d', 'd'],
+  ['e', 'e'],
+  ['f', 'f'],
+  ['g', 'g'],
+  ['h', 'h'],
```

Comment:
I would recommend that we add at least one phonetic alphabet, like the [NATO phonetic alphabet](https://en.wikipedia.org/wiki/NATO_phonetic_alphabet). You can see in the non visual UI we also use [a few different phonetic alphabets](https://github.com/brollin/lila/blob/9549f9c179acf09cd484e4746c3bfb95a9dd6809/ui/nvui/src/chess.ts#L23-L42). I suspect that we will get misrecognition complaints no matter what we do here, given how hard it is for a language model to accurately recognize different accents, along with how frustrating it is to play the wrong move sometimes. Luckily, I think we could get away with adding many phonetic alphabets without problem.

For some more background, in playing around with this stuff in my own branch back in the day, when saying "E4", it would often be misrecognized as "A4". Also, the number "8" would sometimes be heard as the letter "A". And that is all with a high quality microphone very close to my mouth in a quiet environment, which the average player won't have. I think a help page with tips for minimizing misrecognitions will probably be in order, and I'm happy to help with such a thing.

Reply:
I'll add the phonetic alphabet. If you get the chance, this branch is live on https://schlawg.org if you create an account, log in and enable voice input in settings. I test it on puzzles

Reply:
Very nice, tried it out there and it works quite well! 🎉  Here are some quick notes:

* There were very few misrecognitions overall, a great sign! When they did occur, I got the red keyboard move box but no other feedback. It would be nice if it displayed what it thought I said. This helps so that you know which word was misheard and can attempt to be clearer pronouncing it.
* As noted elsewhere, red for off might not be ideal. Maybe a gray microphone for off? Also making it more clearly a button might be nice, or maybe just making it larger.
* When it is on, having a light animation could be good, something to remind people that they are being listened to. (as a side note, I love that we are doing the voice recognition in the browser, so that we can explain elsewhere that we are not sending recordings anywhere)
* In general, code-wise it would be great to separate the voice stuff from the keyboardMove stuff. The overlapping concerns could make for tricky maintenance going forward. Definitely easier said than done - I know that keyboard move already has its move logic somewhat coupled with the presentation layer. My drive by thought here is that maybe in the end we have 3 modules - keyboardMove, voiceMove, moveLogic. The last one I am the most unsure about (esp the name), but I imagined it as the logic common to both of keyboardMove and voiceMove. But may be all of that isn't necessary, just my quick thoughts. Happy to chat more about that if it would be helpful.